RELEASE_NOTE_DRAFT.md

```diff
@@ -1,37 +1,108 @@
+* Changed collection constructor factories from enums to singleton classes.
+* Upgraded CollectPrimitiveProcedure to be generated at build time by using stg file.
+* Added SerializationTest for CollectPrimitiveProcedure.
+* Added test coverage for IntInterval, MultiReaderHashBag, CharAdapter, CodePointAdapter, CodePointList, Lists, Primitive Factories, RichIterable.minOptional(), RichIterable.maxOptional(), RichIterable.zip(), MapIterable.detectOptional, MapIterable.detectWithOptional and MutableMapIterable.removeIf().
+* Updated IntListJMHTest to include benchmarks using IntStream.
+* Reduced benchmark errors by consuming every object created in a benchmark to avoid DCE.
+
+# Removed Functionality
+-----------------------
+* Removed unused ant build module.
+* Removed deprecated classes marked for removal: SummaryStatistics, SerializableIntSummaryStatistics, SerializableLongSummaryStatistics, and SerializableDoubleSummaryStatistics.
+
+# Build Changes
+-----------------
+* Added Javadoc build for pull requests to ensure correct CI.
+* Upgraded maven plugins for build time stability.
+* Updated Tycho Version to 1.3.0.
+* Configured Travis to run maven using the maven wrapper.
+* Fixed 'maven install' for newer versions of Java by using maven profiles to link modules with the version of Java that can build them.
+* Fixed maven pluginRepositories settings to include maven central.
+* Fixed Travis builds to cache the maven local cache without including Eclipse Collections jars.
+* Removed EclipseCollectionsCodeGenerator Task.
+* Removed unused dependencies from maven build.
+* Removed obsolete maven prerequisites.
+* Customized IntelliJ maven argline settings.
+* Turned on Eclipse Collections Code Generator plugin on configuration for Eclipse IDE.
+
+# Breaking Changes
+------------------
+_Warning_: These changes are already mentioned above. The list below might not be exhaustive, make sure to test your application and usages to verify.
+* Changed groupByUniqueKey() so that it's target extends MutableMapIterable instead of MutableMap. This breaks binary compatibility.
+* Added missing bounded wildcards on generic types.
+* Removed deprecated classes marked for removal: SummaryStatistics, SerializableIntSummaryStatistics, SerializableLongSummaryStatistics, and SerializableDoubleSummaryStatistics.
+* Removed default implementation for RichIterable.toSortedMapBy().
+* Removed default implementation for selectUnique() from all Bag interfaces viz. Bag, ImmutableBag, ImmutableBagIterable, ImmutablePrimitiveBag, ImmutableSortedBag, MutableBag, MutableBagIterable, MutablePrimitiveBag, MutableSortedBag, PrimitiveBag, SortedBag, UnsortedBag.
+* Removed default implementation for MutableListFactory.ofInitialCapacity().
```

Comment:
Removed redundant default impl .... existed only for backward compatibility

Reply:
fixed